yt_dlp/extractor/ard.py

```diff
@@ -360,11 +361,30 @@ def _extract_episode_info(self, title):
     def _real_extract(self, url):
         display_id = self._match_id(url)
 
+        query = {
+            'embedded': 'false',
+            'mcV6': 'true'
+        }
+        headers = {}
+
+        ams_cookie = self._get_cookies("https://sso.ardmediathek.de/sso/token").get("ams")
+        if ams_cookie:
+            token = self._download_json('https://sso.ardmediathek.de/sso/token', display_id,
+                                        fatal=False, note='Getting token for age verification')
+            id_token = traverse_obj(token, 'idToken')
+            if not id_token:
+                self.report_warning('Unable to find id token, continuing without authentication')
+
+            else:
+                jwt_token_decoded = jwt_decode_hs256(id_token)
+                user_id = traverse_obj(jwt_token_decoded, 'user_id', 'sub')
+
+                headers['x-authorization'] = f'Bearer {id_token}'
+                query['userId'] = user_id
```

Comment:
using the power of traversal, we can check both values before warning/aborting

just to confirm, the `user_id` is found in both the `user_id` and `sub` fields of the JWT, right?
```suggestion
            id_token = traverse_obj(token, ('idToken', {str}))
            user_id = traverse_obj(id_token, ({jwt_decode_hs256}, ('user_id', 'sub'), {str}), get_all=False)
            if not id_token or not user_id:
                self.report_warning('Unable to extract token, continuing without authentication')
            else:
                headers['x-authorization'] = f'Bearer {id_token}'
                query['userId'] = user_id
```

Reply:
yeah exactly, see also my investigation in the issue:
https://github.com/yt-dlp/yt-dlp/issues/9035#issuecomment-1902065076